### PR TITLE
NIFI-2708 Added merging of SiteToSite details

### PR DESCRIPTION
**Pull request by jtstorck**

### CODE REVIEW

**Comment by jtstorck** [comment]
Ready for review.

**Comment by mcgilman** [comment]
Reviewing...

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-cluster/src/main/java/org/apache/nifi/cluster/coordination/http/endpoints/ControllerEndpointMerger.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        }
+
+        // merge output ports
+        for (Map<NodeIdentifier, PortDTO> outputPortByNodeId : outputPortMap.values()) {
+            final Collection<PortDTO> nodeOutputPorts = outputPortByNodeId.values();
+            if (!nodeOutputPorts.isEmpty()) {
+                final PortDTO outputPort = nodeOutputPorts.iterator().next();
+                final PortDTO clientOutputPort = clientDto.getInputPorts().stream().filter(p -> p.getId().equals(outputPort.getId())).findFirst().orElse(null);
+                if (clientOutputPort != null) {
+                    PortEntityMerger.mergeDtos(clientOutputPort, outputPortByNodeId);
```
Comment (mcgilman) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-cluster/src/main/java/org/apache/nifi/cluster/coordination/http/endpoints/ControllerEndpointMerger.java]: It looks like `PortEntityMerger.mergeDtos` is expecting the `dtoMap` (2nd argument) to contain all DTOs including the `clientDto`. In the `ControllerEndpointMerger` it looks like the `dtoMap` will contain all DTOs except the `clientDto`. As a result, it looks like any `validationErrors` in `clientDto` will be lost.

**Comment by jtstorck** [comment]
Good catch.  Updated so that it doesn't skip over aggregating the clientDto ports.

**Comment by mcgilman** [comment]
Looks good @jtstorck! This has been merged to master.

**Pull request closed without merging**